kernel/src/process_standard.rs

```diff
@@ -258,6 +276,64 @@ impl<C: Chip> Process for ProcessStandard<'_, C> {
         ret
     }
 
+    /// Transition an unverified process into the State::Unstarted state.
+    /// Should require a capability. -pal
+    fn mark_credentials_pass(
+        &self,
+        credentials: Option<TbfFooterV2Credentials>,
+        short_app_id: Option<ShortID>,
+        _capability: &dyn capabilities::ProcessApprovalCapability,
+    ) -> Result<(), ErrorCode> {
+        if self.state.get() != State::Unchecked {
+            Err(ErrorCode::NODEVICE)
+        } else {
+            self.state.update(State::Unstarted);
+            self.app_id.insert(short_app_id);
+            credentials.map(|c| self.credentials.replace(c));
+            Ok(())
+        }
+    }
+
+    fn mark_credentials_fail(&self, _capability: &dyn capabilities::ProcessApprovalCapability) {
+        self.state.update(State::CredentialsFailed);
+    }
+
+    fn get_credentials(&self) -> Option<TbfFooterV2Credentials> {
+        let c = self.credentials.take();
+        self.credentials.insert(c);
+        c
+    }
+
+    /// Enqueue the initialization function of a process onto its task list;
+    /// this is used to start a process. Should only be called when a process
+    /// is in the `State::Unstarted` state.
+    fn enqueue_init_task(
+        &self,
+        _cap: &dyn capabilities::ProcessInitCapability,
+    ) -> Result<(), ErrorCode> {
+        if self.state.get() != State::Unstarted && self.state.get() != State::Terminated {
+            return Err(ErrorCode::NODEVICE);
+        }
+
+        self.state.update(State::Yielded);
+
+        // And queue up this app to be restarted.
+        let flash_start = self.flash_start();
+        let app_start =
+            unsafe { flash_start.offset(self.header.get_app_start_offset() as isize) as usize };
+        let init_fn =
+            unsafe { flash_start.offset(self.header.get_init_function_offset() as isize) as usize };
```

Comment:
Use `wrapping_offset` so we can avoid the `unsafe`.

Reply:
This was copied from

https://github.com/tock/tock/blob/1d2094012a779dbe7b253725e710f0a177bfa668/kernel/src/process_standard.rs#L1341

But yeah I'll change it. Done.